dns/r53/client.go

```diff
@@ -47,24 +52,35 @@ func (dns *client) GetHostedZoneDomain() (string, error) {
 }
 
 // UpdateRecordSets updates records in aws based on the change list.
-// Todo add tests
 func (dns *client) UpdateRecordSets(changes []*route53.Change) error {
-	recordSetsInput := &route53.ChangeResourceRecordSetsInput{
-		HostedZoneId: aws.String(dns.hostedZone),
-		ChangeBatch: &route53.ChangeBatch{
-			Changes: changes,
-		},
-	}
+	for i := 0; i < len(changes); i += dns.maxRecordChanges {
```

Comment:
Perhaps that is a confusing name for the self reference. It isn't qualified it is a member on the struct so I could override it for testing.